### PR TITLE
CC-12825: Fix proprietary dependencies

### PR DESCRIPTION
**Pull request by dosvath**
## Problem
Integration tests introduced proprietary dependencies. 

## Solution
Adjust integration tests to not depend on proprietary packages in the S3 sink connector.

Temporarily delete the `S3SinkConnectorNetworkIT` class while a replacement container to `SuidProxy` is found, ticket to track work was created. 

##### Does this solution apply anywhere else?
- [ ] yes
- [X] no

##### Testing done:
- [X] Unit tests
- [X] Integration tests
- [ ] System tests
- [ ] Manual tests

## Release Plan

 
Backporting to `5.5.x` where ITs were introduced.

### CODE REVIEW

**Comment by grantatspothero** [comment]
Thanks, was able to build this locally!

**Comment by grantatspothero** [comment]
When trying to run tests I was running into scala problems due to usage of the scala library `io.findify.s3mock`
```
java.lang.NoClassDefFoundError: scala/Serializable
```

It looks like this old branch does not include scala as a test dependency while the master branch does:
https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/pom.xml#L139-L144

**Comment by dosvath** [comment]
> When trying to run tests I was running into scala problems due to usage of the scala mocking library `io.findify.s3mock`
> 
> ```
> java.lang.NoClassDefFoundError: scala/Serializable
> ```
> 
> It looks like this old branch does not include scala as a test dependency while the master branch does:
> https://github.com/confluentinc/kafka-connect-storage-cloud/blob/master/kafka-connect-s3/pom.xml#L139-L144

That sounds correct, with https://github.com/confluentinc/kafka-connect-storage-cloud/pull/329 this was included in master but not backported to `5.5.x`. Once we merge this PR, these changes will be ported to master so it will include both fixes.

**Comment by grantatspothero** [comment]
Ahh perfect, did not realize you were backporting this to 5.5.x 👍

File: kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/README.md [hunk 1]
```diff
@@ -9,4 +9,8 @@ the `.aws/credentials` file.
 
 ## Running the Tests
 Tests can be run manually using the `mvn integration-test` command 
-or through IntelliJ by running the class.
\ No newline at end of file
+or through IntelliJ by running the class.
+
+If running in IntelliJ, temporarily disable the network ITs in by
```
Comment (kkonstantine) [inline kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/README.md:14]: See my comment above. 
That's an annoying hoop to jump

**Review by kkonstantine — COMMENTED** [review]
Thanks for the quick turnaround @dosvath 
Overall looks good. 

I have a couple of comments. Mainly, we'll have to consider removing the code that doesn't run ootb.

File: kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/README.md (hunk 1 above)
Comment (dosvath) [reply kafka-connect-s3/src/test/java/io/confluent/connect/s3/integration/README.md]: Agree

File: kafka-connect-s3/pom.xml [hunk 2]
```diff
@@ -112,13 +98,13 @@
         </dependency>
         <dependency>
             <groupId>org.apache.kafka</groupId>
-            <artifactId>kafka_${kafka.scala.version}</artifactId>
```
[the remark this answers is not in the document]
Comment (dosvath) [reply kafka-connect-s3/pom.xml:115]: Linked to detailed discussion thread for this offline, we start having issues with the `EmbeddedConnectCluster` starting `2.13`

**Comment by dosvath** [comment]
Updated PR with temporary removal of the test file that does not compile. Thanks @kkonstantine!

**Review by levzem — APPROVED** [review]
LGTM

**Pull request merged**